src/async_client.rs

```diff
@@ -337,7 +382,15 @@ impl AsyncClient {
 
         let response = self.send(request).await?;
         let mut client = self.base_client.write().await;
-        client.receive_login_response(&response).await?;
+        // TODO avoid allocation somehow?
+        let path = self.state_store_path.as_ref().map(|p| {
+            let mut path = PathBuf::from(p);
+            path.push(response.user_id.to_string());
```

Comment:
I can't figure out a good way to make this work without a way to append to the state store path?
[here](https://github.com/DevinR528/RumaTui/blob/master/src/client/mod.rs#L53-L93) is the client creation, and when I login, I forked this to test it out. Sorry for making this more complicated

Reply:
Ah right, we only give the user when we log in. Forgot about that problem 🤦.

Sorry I missed that, we can leave the per user dir creation in the state store as is since the alternative is more complicated.

But we'll need to document this behavior in the `JsonStore` docs.

Reply:
Should I add a `push_path` method to the StateStore trait, or make each take a path again? The push_path method has the advantage that it doesn't muck up the load/store functions for everyone else impling StateStore, we could make it a no-op default method, so the user doesn't have to impl it if it doesn't make sense for the user's client.

Reply:
I don't think a `push_path` method is the right answer here, in a SQLite context it certainly doesn't make sense.

Can't the `JsonStore` figure out a path if we give it the `Session` object. Ah I see now. You also store/load the `Session` as client state, if we move the responsibility of storing the `Session` to the clients this solves itself, we also accept the `Session` when creating the client so 

Another reason to avoid storing the session, the access token should probably stored somewhere safe, for example the OS key-ring instead of the state store.

So yeah, either the lib user gives us a completely unique folder for a user and we don't create subfolders for users, or the lib user becomes responsible to store the credentials.

If we still want to store the `user_id` and `device_id` we should somehow split that out from the access token, since we really want to avoid being responsible for the access token.

Reply:
Right, if you take a look where the crypto thing gets created, if a `Session` is provided at client creation it's created as well, otherwise at login time.

That way we support the use-case where a client doesn't remember the access token, but remembers the password instead. I think the same thing makes sense for the state store. You either remember the `Session` and create a Client object with a loaded store, or you log in again and it'll get the `Session` that way.

Reply:
So we can leave it up to the client user to store the `Session`(specificaly the access_token) but like the `CryptoStore` we store the user_id (and device_id) and use that to add to the `JsonStore` path, yea that's way more elegant!! I hope I'm understanding this correctly.

Reply:
Yeah, the cryptostore uses the user_id/device_id to uniquely store stuff in the database.

The `JsonStore` can do the same but with folders instead of a database.